### PR TITLE
Add unit tests for grpc server

### PR DESCRIPTION
**Pull request by ctlnmhl**
FYI @nanzhong 
- added unit tests following the existing approach

### CODE REVIEW

File: grpc/server/servertest.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package server
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/servertest.go]: fixed!

File: grpc/server/servertest.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+	commonv1 "github.com/nanzhong/tstr/api/common/v1"
+	"github.com/nanzhong/tstr/grpc/types"
+)
+
+type AccessTokenBuilder struct {
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/servertest.go]: fixed!

File: grpc/server/admin_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+func TestAdminServer_GetAccessToken(t *testing.T) {
+	tests := []struct {
+		name    string
+		token   *commonv1.AccessToken
+		errCode codes.Code
+		errMsg  string
+	}{
+		{
+			"valid acces token uuid in the request",
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: fixed!

File: grpc/server/admin_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			NewAccessTokenBuilder().WithRevokedAt().Build(),
+			codes.OK,
+			"",
+		},
+		{
+			"invalid acces token uuid in the request, multiple scopes",
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: initially I planned on using a combination of scopes but it didn't seem to be relevant for the testing
removed the scopes part from test description

File: grpc/server/admin_test.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+				request = &adminv1.GetAccessTokenRequest{Id: tokenID.String()}
+			}
+
+			res, err := server.GetAccessToken(ctx, request)
+			if res != nil {
+				require.NoError(t, err)
+				assert.Equal(t, &adminv1.GetAccessTokenResponse{AccessToken: tt.token}, res)
+			} else {
+				if er, ok := status.FromError(err); ok {
+					if er.Code() != tt.errCode {
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: done!

File: grpc/server/admin_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: agree. fixed!

File: grpc/server/identity_test.go [hunk 6]
```diff
@@ -35,14 +32,10 @@ func TestIdentityServer_Identity(t *testing.T) {
...
-		Id:                 tokenID.String(),
-		Name:               "name",
-		NamespaceSelectors: []string{"ns-0"},
-		Scopes:             []commonv1.AccessToken_Scope{commonv1.AccessToken_ADMIN},
-		IssuedAt:           types.ToProtoTimestamp(time.Now()),
-		ExpiresAt:          types.ToProtoTimestamp(time.Now().Add(time.Hour)),
+	token := NewAccessTokenBuilder().Build()
+	tokenID, err := uuid.Parse(token.Id)
+	if err != nil {
+		t.Skip("unable to parse the token id", token.Id)
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/identity_test.go]: ack! switched from Skip to Error

File: grpc/server/admin_test.go [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	tokenString := "token"
+	ctx := metadata.NewIncomingContext(context.Background(), metadata.Pairs(auth.MDKeyAuth, "bearer "+tokenString))
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var request *adminv1.GetAccessTokenRequest
+			if tt.token != nil {
+				tokenID, err := uuid.Parse(tt.token.Id)
+				if err == nil {
+					mockQuerier.EXPECT().GetAccessToken(ctx, gomock.AssignableToTypeOf(server.pgxPool), tokenID).Return(db.GetAccessTokenRow{
```
[the remark this answers is not in the document]
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: `func(*db.MockQuerier)` would have been required too many arguments
using the same approach, I'm returning only the output for the mockQuerier which only needs `*commonv1.AccessToken` as argument

File: grpc/server/admin_test.go [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+func TestAdminServer_GetAccessToken(t *testing.T) {
+	tests := []struct {
+		name              string
+		token             *commonv1.AccessToken
+		errCode           codes.Code
+		errMsg            string
+		mockQuerierReturn func(*commonv1.AccessToken) (db.GetAccessTokenRow, error)
+	}{
+		{
+			name:    "valid acces token uuid in the request",
```
Comment (nanzhong) [inline grpc/server/admin_test.go:41]: One more small typo `s/acces/access/` 😀.

File: grpc/server/admin_test.go [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+			mockQuerierReturn: func(token *commonv1.AccessToken) (db.GetAccessTokenRow, error) {
+				tokenID, _ := uuid.Parse(token.Id)
+				return db.GetAccessTokenRow{
+					ID:                 tokenID,
+					Name:               token.Name,
+					NamespaceSelectors: token.NamespaceSelectors,
+					Scopes:             []string{"admin"},
+					IssuedAt:           sql.NullTime{Valid: true, Time: token.IssuedAt.AsTime()},
+					ExpiresAt:          sql.NullTime{Valid: true, Time: token.ExpiresAt.AsTime()},
+					RevokedAt:          sql.NullTime{Valid: true, Time: token.RevokedAt.AsTime()},
```
Comment (nanzhong) [inline grpc/server/admin_test.go:54]: Even though this doesn't affect the test, this doesn't quite line up with how the sql null types work. The `Valid` field here is used to differentiate a null sql value vs a non-null value. You can use https://github.com/nanzhong/tstr/blob/main/grpc/types/from.go#L61 to handle the conversion.

File: grpc/server/server_test.go [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/google/uuid"
+	commonv1 "github.com/nanzhong/tstr/api/common/v1"
+	"github.com/nanzhong/tstr/grpc/types"
+)
+
+type accessTokenBuilder struct {
+	accessToken *commonv1.AccessToken
+}
+
+func NewAccessTokenBuilder() *accessTokenBuilder {
```
Comment (nanzhong) [inline grpc/server/server_test.go:15]: I think this function should be non-exported since it shouldn't be used outside of this package.

**Review by nanzhong — CHANGES_REQUESTED** [review]
Looking better! Just a few small comments and it should be good to merge after.

File: grpc/server/server_test.go (hunk 10 above)
Comment (ctlnmhl) [reply grpc/server/server_test.go]: ack!

File: grpc/server/admin_test.go (hunk 9 above)
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: ack! Updated!

File: grpc/server/admin_test.go (hunk 8 above)
Comment (ctlnmhl) [reply grpc/server/admin_test.go]: https://media.giphy.com/media/2wSaulb0fsDydh0IoB/giphy-downsized-large.gif

Comment (ctlnmhl) [reply grpc/server/admin_test.go]: https://media.giphy.com/media/2wSaulb0fsDydh0IoB/giphy-downsized-large.gif

**Review by nanzhong — APPROVED** [review]
Thanks, lgtm. Rebasing and will merge on green.

**Pull request merged**